sdk/identity/Azure.Identity/src/KubernetesProxyHttpHandler.cs

```diff
@@ -0,0 +1,271 @@
+            }
+            else
+            {
+                combinedPath = proxyPath + "/" + requestPath;
+            }
+
+            // Build URI string directly to avoid UriBuilder.Path escaping
+            string uriString = $"{newUri.Scheme}://{newUri.Host}:{newUri.Port}{combinedPath}";
+            request.RequestUri = new Uri(uriString);
+            request.Headers.Host = _config.SniName;
+        }
+
+        private bool ShouldReloadHandler()
+        {
+            byte[] currentCaData = GetCaCertificateData();
+
+            // Keep using current handler during rotation gaps (empty or missing file)
+            if (currentCaData == null || currentCaData.Length == 0)
+            {
+                return false;
+            }
+
+            if (_lastCaData == null || !AreByteArraysEqual(currentCaData, _lastCaData))
+            {
+                return true;
+            }
+
+            return false;
+        }
+
+        private byte[] GetCaCertificateData()
+        {
+            try
+            {
+                string caContent = _config.GetCaCertificateContent();
+                if (string.IsNullOrEmpty(caContent))
+                {
+                    return null;
+                }
+
+                return System.Text.Encoding.UTF8.GetBytes(caContent);
+            }
+            catch
+            {
+                return null;
+            }
+        }
+
+        private static bool AreByteArraysEqual(byte[] a, byte[] b)
+        {
+            if (a == null && b == null)
+                return true;
+            if (a == null || b == null)
+                return false;
+            if (a.Length != b.Length)
+                return false;
+
+            for (int i = 0; i < a.Length; i++)
+            {
+                if (a[i] != b[i])
+                    return false;
+            }
+
+            return true;
+        }
+
+        private DisposingHttpClientHandler CreateHandler()
+        {
+            var handler = new DisposingHttpClientHandler();
+
+            byte[] caData = GetCaCertificateData();
+            _lastCaData = caData;
+
+            // Configure TLS/SSL settings if custom CA is provided
+            if (caData != null && caData.Length > 0)
+            {
+                ConfigureCustomCertificates(handler, caData);
+            }
+
+            return handler;
+        }
+
+        private void ConfigureCustomCertificates(HttpClientHandler handler, byte[] caData)
+        {
+            try
+            {
+                string pemContent = System.Text.Encoding.UTF8.GetString(caData);
+                X509Certificate2 caCertificate = PemReader.LoadCertificate(pemContent.AsSpan(), allowCertificateOnly: true);
+
+                handler.ServerCertificateCustomValidationCallback = (request, cert, chain, errors) =>
+                {
+                    if (errors == System.Net.Security.SslPolicyErrors.None)
+                    {
+                        return true;
```

Comment:
Here's my naive observation as someone unfamiliar with this API: the docs for [SslPolicyErrors](https://learn.microsoft.com/en-us/dotnet/api/system.net.security.sslpolicyerrors?view=net-9.0) imply the caller has performed only very basic checks, which seems to imply this function will generally return true here without checking anything else